src/operator/tensor/matrix_op.cu

```diff
@@ -40,7 +40,8 @@ NNVM_REGISTER_OP(_backward_slice_axis)
 .set_attr<FCompute>("FCompute<gpu>", SliceAxisGrad_<gpu>);
 
 NNVM_REGISTER_OP(dot)
-.set_attr<FCompute>("FCompute<gpu>", DotForward_<gpu>);
+.set_attr<FCompute>("FCompute<gpu>", DotForward_<gpu>)
+.set_attr<FComputeEx>("FComputeEx<gpu>", DotForwardEx<gpu>);
 
```

Comment:
Good point. I think gpu is tested when default_ctx is set to a gpu context when test_operator_gpu.py is run. We should add "from test_sparse_operator import *" to that file.